src/System.Net.Http/src/System/Net/Http/MessageProcessingHandler.cs

```diff
@@ -12,6 +13,22 @@ namespace System.Net.Http
 {
     public abstract class MessageProcessingHandler : DelegatingHandler
     {
+        // Private class used to capture the SendAsync state in
+        // a closure, while simultaneously avoiding a tuple allocation.
+        private class SendState : TaskCompletionSource<HttpResponseMessage>
+        {
+            public MessageProcessingHandler self;
+            public CancellationToken token;
```

Comment:
Nit: `readonly` and the names should be capitalized. Could also be `internal` (if you change these properties to internal, I'd change the constructor as well).